src/test/rmc/Match/match_bool.rs

```diff
@@ -0,0 +1,28 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+// compile-flags: --crate-type lib
+// rmc-flags: --function match_bool
+
+#[rmc::proof]
+pub fn match_bool(arg: bool) {
```

Comment:
I wouldn't use proof annotations on functions with arguments. This isn't really supported (yet!) and only sometimes sorta works.

Reply:
I kind of prefer using the proof annotation than the `#[no_mangle]`. What are the downsides to it?

Reply:
I'm just suggesting switching from an argument to `let arg = rmc::nondet();` at the start.

Reply:
I misunderstood you. Sorry about that. Sure, I can do that